### PR TITLE
Guide updated after ADC refactoring

### PR DESCRIPTION
**Pull request by pmarcinkiew**
This change updated guide to match new APIs implementations started with ADC https://github.com/Samsung/iotjs/pull/1012

### CODE REVIEW

**Comment by pmarcinkiew** [comment]
Duplicated

**Comment by yichoi** [comment]
https://travis-ci.org/Samsung/iotjs/jobs/252039819

**Comment by yichoi** [comment]
@pmarcinkiew you are supposed to fix this first 
```
tools/check_signed_off.sh --travis
Signed-off-by message is incorrect. The following line should be at the end of the 827ff35 commit's message: 'IoT.js-DCO-1.0-Signed-off-by: Piotr Marcinkiewicz p.marcinkiew@samsung.com'. 
```

**Comment by pmarcinkiew** [comment]
I have some trouble with overwriting commit message so I will recommit this change again.

File: docs/devs/Extended-API-Guidelines.md [hunk 1]
```diff
@@ -9,15 +9,18 @@ However, extended APIs need a guideline because they are implemented by many con
 2. Basically, all APIs are async API. If you want to make sync API, you need to add `Sync` as a suffix.<br>For example, `readSync()`, `writeSync()`, and so on.
 
 ## Generating an object 
-1. The module object should be generated using `open()` API for consistent usability. (Do not use `New` constructor)
-2. `open()` API should have configuable as first argument and callback function as second argument.<br>callback function is always optional.
+1. The module object should be generated using `new' API for consistent usability.
+2. `new` API should have configuable as first argument and callback function as second argument.<br>callback function is always optional.
```
Comment (hs0225) [inline docs/devs/Extended-API-Guidelines.md]: I think, `Constructor` is better than `new API`.

**Comment by yichoi** [comment]
no more updates?

**Comment by pmarcinkiew** [comment]
Would you be so kind to review my changes again?

**Comment by pmarcinkiew** [comment]
Would you be so kind to review my changes again?

I tested this commit again using test_adc.js with the latest TizenRT and it passed:

```
TASH>>iotjs test_adc.js
adc_setup: Enable the ADC interrupt: irq=106
ADC initialized
test1 start(read async test)
4008
4003
4007
4002
4007
test1 complete
adc_setup: Enable the ADC interrupt: irq=106
ADC initialized
test2 start(read sync test)
3657
3650
3650
3651
3653
test2 complete
iotjs thread end
TASH>>
```

I use potentiometer connected to ground and 3.3 VC from Artik053. The middle connector is providing signal to ADC0 input. I changed resistance for sync part of test and it is clear that output changes.

**Pull request merged**